testaddons/tests.go

```diff
@@ -269,7 +598,54 @@ func (options *TestAddonOptions) RunAddonTest() error {
 		options.Logger.ShortInfo("Finished PreDeployHook")
 	}
 
-	// Trigger Deploy
+	// validate if expected dependencies are deployed for each addon
+	if !options.SkipDependencyValidation {
+		options.Logger.Info("Starting with dependency validation")
+		var rootCatalogID, rootOfferingID, rootVersionLocator string
+		for _, config := range allConfigs {
+
+			if *config.ID == configDetails.ConfigID {
+				rootVersionLocator = *config.Definition.LocatorID
+				break
+			}
+		}
+		// get catalog ID from version locator
+		rootCatalogID = strings.SplitN(rootVersionLocator, ".", 2)[0]
+		rootOfferingID = options.AddonConfig.OfferingID
+
+		graph := make(map[cloudinfo.OfferingNameVersionFlavor][]cloudinfo.OfferingNameVersionFlavor)
+		visited := make(map[string]bool)
+		err = options.buildDependencyGraph(rootCatalogID, rootOfferingID, rootVersionLocator, options.AddonConfig.OfferingFlavor, graph, visited)
+		if err != nil {
+			return err
+		}
+		visited2 := make(map[string]bool)
+
+		actuallyDeployedList := make([]cloudinfo.OfferingNameVersionFlavor, 0)
+
+		options.Logger.ShortInfo("printing the actually deployed configs")
+
+		options.buildactuallydeployedList(options.AddonConfig, visited2, &actuallyDeployedList)
+
+		for key, value := range graph {
+
+			options.Logger.Info(fmt.Sprintf("{%s %s %s} needs ", key.Name, key.Version, key.Flavor))
+
+			for _, dep := range value {
+
+				options.Logger.ShortInfo(fmt.Sprintf("{%s %s %s} ", dep.Name, dep.Version, dep.Flavor))
+			}
+			fmt.Println()
+		}
+
+		// now validate what is actually deployed by iterating over expected dependency graph and actually deployed List
+		err = options.validateDependencies(graph, actuallyDeployedList)
```

Comment:
It is not failing if there are extra dependencies that were not expected
![image](https://github.com/user-attachments/assets/749ba793-43a6-4b29-ac73-931e52024593)

We should print the expected list of configs and the actual list to help with debugging